### ISSUE TITLE
loadjson sensitive to tabs in whitespace?

### ISSUE THREAD

**Issue by lebedov** [fangq/jsonlab#30]
It seems that the presence of tabs in a JSON file can cause `loadjson()` to load certain files differently. Using revision f8def3d of jsonlab with either MATLAB R2016a or Octave 4.2.0, loading `foo.json` from [files.zip](https://github.com/fangq/jsonlab/files/674353/files.zip) (which contains tabs) using `data = loadjson('foo.json')` causes `data.a.b.c` to be created as a cell array, while loading `bar.json` (which contains only spaces) causes `data.a.b.c` to be created as a matrix. Shouldn't `c` always be created as a matrix regardless of the whitespace?

**Comment by fangq** [comment]
@lebedov, thanks for reporting this issue. I debugged loadjson with both of your provided json files, and found the issue was caused by `sscanf` when parsing trailing whitespace in an array construct. for example:

[obj, count, errmsg, nextidx]=sscanf(sprintf('1,2,3'),'%f,',[1,inf])

and 

[obj, count, errmsg, nextidx]=sscanf(sprintf('1,2,3\t'),'%f,',[1,inf])

return different errmsg and nextidx outputs. I added one line to remove the trailing white space, and now both inputs generate the same array. let me know if there is any problem with this fix.
